pkg/sql/codegen/tensorflow/codegen.go

```diff
@@ -209,6 +211,20 @@ func Train(trainIR *ir.TrainClause) (string, error) {
 			fmt.Sprintf("\"%s\": [%s]", target, strings.Join(perTargetFeatureColumnsCode, ",\n")))
 	}
 	isKeras, estimatorStr := IsKerasModel(trainIR.Estimator)
+	isPAI := os.Getenv("SQLFLOW_submitter") == "pai"
+	paiTable := ""
+	// TODO(typhoonzero): if isPAI, create two Couler steps
+	if isPAI {
+		fromRegex, err := regexp.Compile("FROM\\s([a-z0-9_\\.]*)")
+		if err != nil {
+			return "", err
+		}
+		matches := fromRegex.FindAllStringSubmatch(trainIR.Select, -1)
+		if len(matches) != 1 {
+			return "", fmt.Errorf("only support simple SQL query, but got %s", trainIR.Select)
```

Comment:
Can't we utilize the third party SQL parser to support complex SQL queries here?

Reply:
No. Queries with subqueries will not write the result to a table, the result is produced in memory.